web/pages/feeds/import.svelte

```diff
@@ -0,0 +1,134 @@
+    handleImportFile,
+    importedCategories,
+    importedFeeds,
+    importedFeedsByCategory,
+    importErrors,
+    reading,
+    selectAllImportedFeeds,
+    submitImport,
+    submiting,
+    importMessages as t,
+    toggleImportedCategory,
+    toggleImportedFeed,
+    unLoadedFeeds
+  } from '@slowreader/core'
+
+  import Button from '../../ui/button.svelte'
+  import Card from '../../ui/card.svelte'
+  import Loader from '../../ui/loader.svelte'
+  import Page from '../../ui/page.svelte'
+  import RadioField from '../../ui/radio-field.svelte'
+  import FeedList from './list.svelte'
+
+  let currentFeeds: 'all' | 'select' = 'all'
+
+  function handleFileChange(e: Event): void {
+    let target = e.target as HTMLInputElement
+    let file = target.files?.[0]
+    if (file) {
+      handleImportFile(file)
+    }
+  }
+
+  function handleRadioChange(e: CustomEvent<'all' | 'select'>): void {
+    currentFeeds = e.detail
+    if (currentFeeds === 'all') {
+      selectAllImportedFeeds()
+    }
+  }
+
+  async function handleSubmit(): Promise<void> {
+    await submitImport()
+  }
+</script>
+
+<Page title={$t.importTitle}>
+  <div class="feeds-import">
+    <div class="feeds-import_hero">
+      <Card>
+        <h2>{$t.importTitle}</h2>
+        <input disabled={$reading} type="file" on:change={handleFileChange} />
+        {#if $reading}
+          <Loader />
+        {/if}
+        {#if $importErrors.length}
+          <h4>Error Messages</h4>
+          <ul>
+            {#each $importErrors as error (error)}
+              <li>{error}</li>
+            {/each}
+          </ul>
+        {/if}
+      </Card>
+    </div>
+
+    {#if $unLoadedFeeds.length}
+      <h4>{$t.loadError}</h4>
+      <ul class="feeds-import_unloaded">
+        {#each $unLoadedFeeds as feed (feed)}
+          <li>
+            {feed}
+          </li>
+        {/each}
+      </ul>
+    {/if}
+
+    {#if $importedFeedsByCategory.length}
+      <form on:submit|preventDefault={handleSubmit}>
+        <Card>
+          <RadioField
+            current={currentFeeds}
+            label={$t.type}
+            values={[
+              ['all', $t.allFeeds],
+              ['select', $t.selectFeeds]
+            ]}
+            on:change={e => {
+              handleRadioChange(e)
+            }}
+          />
+        </Card>
+        <FeedList
+          disabled={currentFeeds === 'all'}
+          feedsByCategory={Array.from($importedFeedsByCategory)}
+          selectedCategories={Array.from($importedCategories)}
+          selectedFeeds={Array.from($importedFeeds)}
+          on:toggleCategory={e => {
+            toggleImportedCategory(e.detail.categoryId)
+          }}
+          on:toggleFeed={e => {
+            toggleImportedFeed(e.detail.feedId, e.detail.categoryId)
+          }}
+        />
+
+        <div class="feeds-import_submit">
+          <Button disabled={$submiting} type="submit">Import</Button>
+          {#if $submiting}
+            <Loader />
```

Comment:
The process and be very-very long.

1. We need to show how many feeds was processes already.
2. Do you do it one-by-one? Parallelism can be faster.

But we can move it to another task later, if you want (you already did an amazing and huge work).

Reply:
1) +
2) I don't why, but then i try use Promise.all for this case, only one request is processed successfully, so for now I left the sequential call